awscli/customizations/s3/utils.py

```diff
@@ -357,10 +357,12 @@ def __init__(self, client, date_parser=_date_parser):
         self._client = client
         self._date_parser = date_parser
 
-    def list_objects(self, bucket, prefix=None, page_size=None):
+    def list_objects(self, bucket, prefix=None, page_size=None, request_payer=None):
```

Comment:
I think that I would prefer that we made this more generalized such as an `extra_args` that takes a dictionary of extra arguments that get passed in. That way if we want to pass information like `Delimeter` or `UrlEncoding` it would be easier.